src/battle/healthbar.c

```diff
@@ -1585,51 +1596,48 @@ static s32 UpdateGauge (s32 max, s32 cur, s32 diff, s32 * temp, u8 size, u16 fil
     return final;
 }
 
-static u8 ov16_02268194 (s32 param0, s32 param1, s32 param2, s32 * param3, u8 * param4, u8 param5)
+static u8 FillCells (s32 max, s32 cur, s32 diff, s32 * temp, u8 * cells, u8 size) //UpdateGaugeCells, FillCells
 {
-    int v0;
-    u32 v1;
-    u32 v2, v3;
-    s32 v4;
-
-    v4 = param1 - param2;
+    int cell;
+    u32 corrected, pixels, final;
+    s32 updated = cur - diff;
 
-    if (v4 < 0) {
-        v4 = 0;
-    } else if (v4 > param0) {
-        v4 = param0;
+    if (updated < 0) {
+        updated = 0;
+    } else if (updated > max) {
+        updated = max;
     }
 
-    v1 = 8 * param5;
+    corrected = 8 * size;
```

Comment:
**suggestion:** I'm not sure about these names; It seems to me like `size` should refer to which of the 8-pixel cells of the healthbar is being updated, and `corrected` would point to the offset of that cell.